### PR TITLE
fix/CAL-1321: Calendar event time granularity

### PR DESCRIPTION
**Pull request by anoissi**

### CODE REVIEW

File: calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js [hunk 1]
```diff
@@ -60,7 +60,14 @@
       jInput.val(gj(item).html());
       wx.UICombobox.setSelectedItem(jInput.get(0));
     } else {
-      wx.UICombobox.complete(this, e);
+       var regex=/^(2[0-3])|[01][0-9]:[0-5][0-9]$/;
```
Comment (thomasdelhomenie) [inline calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js]: Please add a comment to explain why this code has been added.

File: calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js [hunk 2]
```diff
@@ -60,7 +60,14 @@
...
+       var inputValue = this.value;
+       if(regex.test(inputValue)){
+        var hidden = gj(this).prev("input")[0];
+        hidden.value = inputValue;
```
Comment (thomasdelhomenie) [inline calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js]: It bypasses completely the logic of the original method UICombobox.complete (https://github.com/exodev/gatein-portal/blob/develop/web/eXoResources/src/main/webapp/javascript/eXo/webui/WebUIExt.js#L189), so for example the Enter key is not considered anymore. When I press Enter the dropdown is not hidden, whereas it was before the fix.
You should probably call wx.UICombobox.complete(this, e); in any case and add the new logic then to handle our specific case.

File: calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (anoissi) [reply calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js]: ^(2[0-3]|[01][0-9]):[0-5][0-9]$ my work better :+1:

File: calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js [hunk 3]
```diff
@@ -60,7 +60,14 @@
       jInput.val(gj(item).html());
       wx.UICombobox.setSelectedItem(jInput.get(0));
     } else {
-      wx.UICombobox.complete(this, e);
+       var regex=/^(2[0-3])|[01][0-9]:[0-5][0-9]$/;
+       var inputValue = this.value;
```
[the remark this answers is not in the document]
Comment (anoissi) [reply calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js]: The variable  used tow times used tow times

[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply calendar-webapp/src/main/webapp/javascript/eXo/calendar/UICombobox.js]: You can use this.value for these 2 times, no need to use another variable.

**Pull request merged**